### ISSUE TITLE
Receiving MachineAuth and login key but not successfully logging on

### ISSUE THREAD

**Issue by oxters168** [SteamRE/SteamKit#654]
First off, I'm using this library in Unity. I've got the logging in working and friends list coming in on my computer. I can go online and offline etc. The problem is when I build to android. I can start logging in, and it asks me for the 2 factor authentication code then sends me the login key and sentry file. But that's as far as it goes, the Steam Client doesn't send me the logged on callback. So it doesn't go online and grab my friends list. I can't seem to figure out why. I tried using different login ids to see if that would make a difference, but still same result. Also if it helps, I am receiving ClientFriendsGroupsList(5553) callback when I log in on android as well, which I think is strange since I don't seem to be fully logging in.

**Comment by oxters168** [comment]
So apparently I'm also receiving ClientIsLimitedAccount(5430) from Steam. I found [this](https://github.com/SteamRE/SteamKit/issues/528) person possibly had a similar issue. They said that they fixed the issue by exposing port 27018. I don't know how to do that, but I am searching now.

**Comment by azuisleet** [comment]
You will likely need to enable DebugLog for CMClient. It's possible you are simply being disconnected, or you are losing your socket.

**Comment by JustArchi** [comment]
Limited account has **nothing** to do with the connection, it's very likely your network, either intentionally (bad connection), or non-intentionally (android doing shenanigans behind your back to save on power).

Like azu said above, enable debug log and you'll see what is going on.

**Comment by oxters168** [comment]
Thank you so much for your help! I just got back to trying to figure this out and your suggestions helped a lot. I enabled DebugLog and found out that an error was occurring when messages were being decompressed. Apparently Android is missing certain libraries, in my case this specific library. So I downloaded [DotNetZip](https://github.com/haf/DotNetZip.Semverd), built zlib, and included it in my project. Then I replaced all places in SteamKit that had "using System.IO.Compression" with "using Ionic.Zlib". Worked like a charm! Thanks again for all the help :D
